e2e/tests/session_steps_test.go

```diff
@@ -251,6 +251,35 @@ func (s *suite) waitForNewBlockEvent(
 	}
 }
 
+// waitForBlockHeight waits for a NewBlock event to be observed whose height is
+// greater than or equal to the target height.
+func (s *suite) waitForBlockHeight(targetHeight int64) {
```

Comment:
Why not reuse `waitForNewBlockEvent(` inside where the `isEventMatchFn` checks the target height?

Reply:
IIRC it's because `isEventMatchFn` receives `abci.Event`s from `newBlockEvent.Data.Value.ResultFinalizeBlock.Events` which do not contain the block height.

Block height is in `newBlockEvent.Data.Value.Block.Header.Height`.

There's probably room for a refactor but this will affect what `isEventMatchFn` receives.